src/main/java/org/wildfly/security/auth/server/MechanismConfiguration.java

```diff
@@ -41,14 +43,19 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class MechanismConfiguration {
+
+    private final MechanismConfiguration parent;
```

Comment:
Still it seems like a stretch, technically speaking, since overall the "outer" configuration is just a map-by-string-with-default.

Relatedly - right now you have a map, in essence, that is equivalent to Map<mech name, Map<host name, Configuration>>.  I wonder if it would make more sense to instead have Map<host name, Map<mech name, Configuration>>... or even to go the way of TLS and have a more open selection method like SSLContextSelector does (to support SNI and ALPN in the face of almost infinite possible desired arrangements) where the mechanism configuration is selected based on host and mechanism name (and any other parameters we would choose).